src/llmcompressor/modifiers/quantization/gptq/base.py

```diff
@@ -270,28 +307,41 @@ def apply_compression(
         forward_pass_use_cache = self.model.config.use_cache
         self.model.config.use_cache = False
 
-        # in non-sequential mode we run calibration through the full model
         # in sequential mode we run calibration up to the first transformer target
         intermediates = run_calibration_forward(
             self.model, dataloader, mask_padding=True
         )
         self.layer_compressors_[0].clear_early_stop()
 
+        # empty cache if not using sequential update
+        if self.sequential_update == UpdateMethod.OFF:
+            del intermediates
+            gc.collect()
+            torch.cuda.empty_cache()
+
         num_layers = len(self.compressible_layers_)
         for idx, layer_compressor in enumerate(self.layer_compressors_):
             logger.info(f"\n===== Compressing layer {idx+1}/{num_layers} " " =====")
 
-            # Prune/quantize using GPTQ
-            if self.sequential_update:
+            if self.sequential_update != UpdateMethod.OFF:
                 # in sequential mode we run the forward pass for each transformer layer
                 # one at a time, caching the intermediate outputs between layers
-                layer_compressor.pre_compress()
```

Comment:
Just echoing my above comment: this is the call that allocates the Hessian, so to save memory during sequential updates we want to leave this in